util/helm/helm.go

```diff
@@ -58,7 +59,26 @@ func (h *helm) Template(name, namespace string, valuesFiles []string, overrides
 	if err != nil {
 		return nil, err
 	}
-	return kube.SplitYAML(out)
+	objs, err := kube.SplitYAML(out)
+	if err != nil {
+		return nil, err
+	}
+	for _, obj := range objs {
+		annotations := obj.GetAnnotations()
+		if annotations == nil {
+			continue
+		}
+		// Argo CD handles CRD installation so it necessary to treat 'crd-install' hook as a normal resource
+		if hookType, ok := annotations[common.AnnotationHelmHook]; ok && hookType == common.HelmHookCRDInstall {
```

Comment:
Helm hooks can be a comma separated list of strings. I don't think in practice it will happen with the `crd-install` hook, but to be safe, the `==` comparison may need to change to check if the `crd-install` is in the list of hooks.

Reply:
Actually can we solve this in a different way? I prefer not to mutate the users' manifests. Can we update the isHelmHook helper to the following:
```
// isHelmHook indicates if the supplied object is a helm hook
func isHelmHook(obj *unstructured.Unstructured) bool {
	annotations := obj.GetAnnotations()
	if annotations == nil {
		return false
	}
	_, ok := annotations[common.AnnotationHelmHook]
	// insert logic to check presence of crd-install and return false if it is present
	return ok
}
```